### ISSUE TITLE
`\usepgfplotslibrary{colorbrewer}` breaks `\@`

### ISSUE THREAD

**Issue by LaTeXer** [pgf-tikz/pgfplots#466]
Normally, when used after a period `\@` marks it as a non-end-of-sentence period. But once you load the `colorbrewer` library, it seems to redefine `\@` such that it simply gobble the space after it. Even stranger, if you *manually* load the TikZ `colorbrewer` before the PGFPlots one, the effect goes away.

MWE:
```
\documentclass{article}

\usepackage{pgfplots}
% \usetikzlibrary{colorbrewer}
\usepgfplotslibrary{colorbrewer}

\begin{document}%
\noindent%
Test. Test. Test. Test.\\%
Test. Test\@. Test.\@ Test.\\%

\noindent%
x\meaning\@ x\\
x\the\csname @m\endcsname x\\

\end{document}
```

![image](https://github.com/pgf-tikz/pgfplots/assets/141879913/c015b37c-a8db-4942-ac9e-aa39930e3309)

and with the manually loaded TikZ library

![image](https://github.com/pgf-tikz/pgfplots/assets/141879913/af88ee4d-a3e5-4139-8010-25c058905686)

**Comment by muzimuzhi** [comment]
Somehow `\usepgfplotslibrary{colorbrewer}` leaves category code of `@` altered (from 12 to 11). Then `\@` ignores spaces right after it, hence `Test.\@ Test.` becomes `Test.\@{}Test.`

```tex
\documentclass{article}
\usepackage{pgfplots}
\pgfplotsset{compat=newest}

\pgferror{\the\catcode`\@} % ! Package pgf Error: 12.
\usepgfplotslibrary{colorbrewer}
\pgferror{\the\catcode`\@} % ! Package pgf Error: 11.

\begin{document}
\noindent
Test. Test.   Test.     Test.\\
Test. Test\@. Test.\@   Test.\\
Test. Test\@. Test.\@{} Test.\\
\end{document}
```

**Comment by muzimuzhi** [comment]
Duplicate of #216, which has been fixed by #354, but unfortunately the fix is not released.

You can add a `\makeatother` right after `\usepgfplotslibrary{colorbrewer}` as a workaround.
